docs/NNCFArchitecture.md

```diff
@@ -5,20 +5,18 @@ Neural Networks Compression Framework is a set of compression algorithms and too
 In essence, all of the compression algorithms present in NNCF do certain manipulations with the data inside the control flow graph of a DNN - be it the process of quantizing the values of an input tensor for a fully connected layer, or setting certain values of a convolutional layer to zero, etc.
 A general way to express these manipulations is by using hooks inserted in specific points of the DNN control flow graph.
 
-### Model control flow graph tracing
+### NNCFGraph
+To abstract away the compression logic from specifics of the backend, NNCF builds an `NNCFGraph` object for each incoming model object to be compressed.
+`NNCFGraph` is a wrapper over a regular directed acyclic graph that represents a control flow/execution graph of a DNN. 
+Each node corresponds to a call of a backend-specific function ("operator").
+It is built both for the original, unmodified model, and for the model with compression algorithms applied (which, in general, may have additional operations when compared to the original model).
+
+### PyTorch-specific
+#### Model control flow graph tracing
 Unlike other frameworks such as TensorFlow, PyTorch does not have an easily accessible graph representation of a model, and thus no way to identify specific points in the control flow graph.
 For this reason NNCF performs tracing of the PyTorch operators, implemented via wrapping the corresponding function and module calls.
 Through this process of tracing, NNCF builds an internal representation of the model graph, which is then supplied as the point of reference for specification and insertion of hooks at proper places in the network.
 
```

Comment:
While reading this design document I found that `NNCFNetwork` is mentioned prior to its definition. It looks that the description of the NNCFNetwork entity should go first.